campaigns_api.go

```diff
@@ -0,0 +1,146 @@
+package createsend
+
+import (
+	"errors"
+	"fmt"
+	"github.com/xitonix/createsend/campaigns"
+	"github.com/xitonix/createsend/campaigns/orderfield"
+	"github.com/xitonix/createsend/internal"
+	"github.com/xitonix/createsend/order"
+	"net/url"
+	"time"
+)
+
+type campaignAPI struct {
+	client internal.Client
+}
+
+func newCampaignAPI(client internal.Client) *campaignAPI {
+	return &campaignAPI{client: client}
+}
+
+func (c *campaignAPI) Create(clientID string, campaign campaigns.CampaignFromUrl) (string, error) {
+	path := fmt.Sprintf("campaigns/%s.json", url.QueryEscape(clientID))
+	var cId string
+	err := c.client.Post(path, &cId, campaign)
+	return cId, err
+}
+
+func (c *campaignAPI) CreateFromTemplate(clientID string, campaign campaigns.CampaignFromTemplate) (string, error) {
+	path := fmt.Sprintf("campaigns/%s/fromtemplate.json", url.QueryEscape(clientID))
+	var cId string
+	err := c.client.Post(path, &cId, campaign)
+	return cId, err
+}
+
+func (c *campaignAPI) SendImmediately(campaignID string, confirmationEmails string) error {
+	return Send(c.client, campaignID, confirmationEmails, "Immediately")
+}
+
+func (c *campaignAPI) ScheduleSend(campaignID string, confirmationEmails string, date time.Time) error {
+	return Send(c.client, campaignID, confirmationEmails, date.Format("2006-01-02 15:04"))
+}
+
+func (c *campaignAPI) Test(campaignID string, previewRecipients []string) error {
+	path := fmt.Sprintf("campaigns/%s/sendpreview.json", url.QueryEscape(campaignID))
+	return c.client.Post(path, nil, struct {
+		PreviewRecipients []string
+	}{
+		PreviewRecipients: previewRecipients,
+	})
+}
+
+func (c *campaignAPI) Summary(campaignID string) (campaigns.CampaignSummary, error) {
+	path := fmt.Sprintf("campaigns/%s/summary.json", url.QueryEscape(campaignID))
+	var cs campaigns.CampaignSummary
+	err := c.client.Get(path, &cs)
+	return cs, err
+}
+
+func (c *campaignAPI) EmailClientUsage(campaignID string) ([]campaigns.EmailClientUsage, error) {
+	path := fmt.Sprintf("campaigns/%s/emailclientusage.json", url.QueryEscape(campaignID))
+	var ecu []campaigns.EmailClientUsage
+	err := c.client.Get(path, &ecu)
+	return ecu, err
+}
+
+func (c *campaignAPI) ListsAndSegments(campaignID string) (campaigns.ListsAndSegments, error) {
+	path := fmt.Sprintf("campaigns/%s/listsandsegments.json", url.QueryEscape(campaignID))
+	var ls campaigns.ListsAndSegments
+	err := c.client.Get(path, &ls)
+	return ls, err
+}
+
+func (c *campaignAPI) Recipients(campaignID string, page int, pageSize int, orderField orderfield.OrderField, orderDirection order.Direction) (campaigns.Recipients, error) {
+	var r campaigns.Recipients
+	if orderField == orderfield.Date {
+		return r, errors.New("date is not a valid order field")
```

Comment:
All the custom errors must me wrapped in a Client Error. See [here](https://github.com/xitonix/createsend/blob/master/clients_api.go#L197) for an example please.